### ISSUE TITLE
Harmony Hub can't connect to ha-bridge

### ISSUE THREAD

**Issue by rourke** [bwssytems/ha-bridge#968]
I've been looking into this for hours, but I'm getting desperate. I have my ha-bridge server setup on port 80. `/api/devices` works. Now I want my Harmony One remote to connect to my bridge, but whatever I do it just doesn't work.

I've enabled `Trace UPNP Calls` and while trying to connect from the Harmony app I don't see any incoming UPNP calls. It does seem to work since my Google Home Mini sends out calls every minute:

```
Jun 26 23:55:38 smartserver java[6668]: 2018-06-26 23:55:38,923 [main] INFO  com.bwssystems.HABridge.upnp.UpnpListener - Traceupnp: send upnp discovery template 1 with response address: 192.168.1.115:80 to address: /192.168.1.114:59559
Jun 26 23:55:39 smartserver java[6668]: 2018-06-26 23:55:39,574 [main] INFO  com.bwssystems.HABridge.upnp.UpnpListener - Traceupnp: send upnp discovery template 2 with response address: 192.168.1.115:80 to address: /192.168.1.114:59559
Jun 26 23:55:40 smartserver java[6668]: 2018-06-26 23:55:40,224 [main] INFO  com.bwssystems.HABridge.upnp.UpnpListener - Traceupnp: send upnp discovery template 3 with response address: 192.168.1.115:80 to address: /192.168.1.114:59559
```

`192.168.1.115` is my ha-bridge, and `192.168.1.114` is my Google Home Mini. If I'm correct it shows UPNP works.

I do have an actual Hue Bridge on IP `192.168.1.125`. I have no idea how UPNP actually works, but could it be possible the Harmony app prioritizes the actual Hue Bridge instead of the ha-bridge?

**Comment by rourke** [comment]
Even with my actual Hue Bridge turned off the Harmony app won't detect the ha-bridge. Strangely the Harmony app says it does detect a Hue Bridge, but I don't see incoming calls.

**Comment by bwssytems** [comment]
If you are not seeing the harmony send messages, that would be what I would look at as far as it's network settings. As you see, the ha-bridge shows all upnp calls.  I would load the upnp tools and see if there is any traffic with those. Also, tcpdump can be used.

**Comment by rourke** [comment]
Thanks for the reply. 

Until yesterday I had the .jar file of ha-bridge running on my Pi 1, and it worked great. Now I'm running the same setup on a dedicated machine with Ubuntu Server 18.04 LTS and it just won't detect it.

I'm also running Home Assistant on the same machine, and it auto-detected ha-bridge right away. Perhaps that says something as well.

I looked into these [upnp tools](http://www.meshcommander.com/upnptools), and see a lot of traffic with the Device Spy tool. When I'm using the Harmony app to connect to the ha-bridge, is the request coming from my phone or the hub? As of now no requests is coming from either devices.

I don't know much about networking, and to be honest, I have no idea where to start looking from here. tcpdump gives so much data. I don't know how to interpret it.

**Comment by rourke** [comment]
I know this could very well have nothing to do with ha-bridge, but I don't know what to do. Can you or anyone else help me out?

Both the Hue bridge (green) and ha-bridge (orange) are showing up in the Device Sniffer:

![image](https://user-images.githubusercontent.com/2820753/41981512-6050dfdc-7a29-11e8-8370-eada5f410df3.png)

But only the actual Hue bridge shows up in the Device Spy:

![image](https://user-images.githubusercontent.com/2820753/41981616-a2885574-7a29-11e8-93a2-c680b90fdae8.png)

I compared them closely and the only difference I can spot is the Hue bridge also has NT `urn:schemas-upnp-org:device:basic:1` type of packages.

**Comment by bwssytems** [comment]
Are there any firewall rules on you ubuntu machine that is preventing access?

**Comment by rourke** [comment]
The firewall is inactive.

> user@server:~$ sudo ufw status
> Status: inactive

**Comment by rourke** [comment]
After reading a bit more into it I just want to verify something. I'm sending out a body packet from my desktop PC (which is as much a part of my network as all other devices) with the following information:

```
M-SEARCH * HTTP/1.1
HOST: 239.255.255.250:1900
MAN: "ssdp:discover"
MX: 2
ST: ssdp:all
```

According to the readme there [should be 3 responses](https://github.com/bwssytems/ha-bridge/blob/master/README.md#upnp-listening). The types are:
`upnp:rootdevice`
`uuid:<id>`
`urn:schemas-upnp-org:device:basic:1`

However, the responses I'm getting are just 3x `uuid:<id>`:

```
NOTIFY * HTTP/1.1
HOST: 239.255.255.250:1900
CACHE-CONTROL: max-age=100
LOCATION: http://192.168.1.115:80/description.xml
SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.19.0
NTS: ssdp:alive
hue-bridgeid: 94DE80FFFE62E3A2
NT: uuid:2f402f80-da50-11e1-9b23-94de8062e3a2
USN: uuid:2f402f80-da50-11e1-9b23-94de8062e3a2
```

To put this visually, below is a screenshot. The M-SEARCH packet is being sent (red), the Hue bridge responds immediately with loads of responses (green), and after a while the ha-bridge just has 3 answers of the same type (orange).

![image](https://user-images.githubusercontent.com/2820753/42160618-24c3afc6-7df8-11e8-9781-0ae3a6600800.png)

Is this all correct or might this be the cause of my problems? Perhaps I'm just shooting in the dark here, but my entire home server setup progress is on halt at the moment.

Also, the device sniffer in the upnp tools I mentioned earlier says if the device is not detectable the SCDP document could not be parsed correctly. Not sure what document that is.

**Comment by bwssytems** [comment]
The device sniffer only gives you some basic info in that screen. Each of those responses as different content and that screen does not show it. I believe you can click on each one and get more info.

But, in the end, the ha-bridge is responding appropriately. It is then up to the device to query for the description.xml next.

**Comment by rourke** [comment]
I fixed it.

I reinstalled Ubuntu and gave first priority to ha-bridge. The Harmony app still didn't pick it up. This made me conclude it wasn't the server setup or any other software runnnig on it. 

I decided to hook the Harmony Hub to my PC. Although the Harmony app didn't show any connected Hue Bridges, the MyHarmony application did show a random Philips Hue with an IP address not from my network. I hit the reconnect button and well, it picked it up straight away without a problem. Even while running it on port 8080.

TL;DR: the Harmony Hub had some sort of faulty Hue bridge that the Harmony app didn't show. The MyHarmony software could reconnect to ha-bridge.

Thanks a lot for this great application and the support you've given me!
